docs(hydrated_bloc): improve `storagePrefix` documentation (#4559)
felangel/bloc: packages/hydrated_bloc/README.md

runApp(MyApp());
```

## Custom Storage Prefix
The `storagePrefix` defines the unique storage namespace for your `HydratedBloc` or `HydratedCubit`.

> By default, it uses `runtimeType`, which isn't resilient to obfuscation or minification in production. If `runtimeType` changes, your app loses saved state. This is critical for web apps, where code changes frequently alter the minified runtimeType.

Reviewer: ```suggestion
By default, it uses `runtimeType`, which isn't resilient to obfuscation or minification in production. If `runtimeType` changes, your saved state will be lost. This is especially relevant for web apps, where code changes frequently alter the minified `runtimeType`.
```

Always override `storagePrefix` in production for stable, persistent storage.

```dart
class CounterCubit extends HydratedCubit<int> {
  CounterCubit() : super(0);

  void increment() => emit(state + 1);

  @override
  String get storagePrefix => 'counter_cubit';

  @override
  int fromJson(Map<String, dynamic> json) => json['value'] as int;

  @override

>   Map<String, int> toJson(int state) => { 'value': state };

Reviewer: Nit I don't know if it's useful to include `increment` and `to/fromJson` here since this is specifically focusing on overriding `storagePrefix`. I'd probably omit everything except for the `storagePrefix` override.

Author: I removed everything except the constructor and storage prefix override.
